pkg/apis/apps/v1alpha1/workloadrebalancer_types.go

```diff
@@ -0,0 +1,127 @@
+/*
+Copyright 2024 The Karmada Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+const (
+	// ResourceKindWorkloadRebalancer is kind name of WorkloadRebalancer.
+	ResourceKindWorkloadRebalancer = "WorkloadRebalancer"
+	// ResourceSingularWorkloadRebalancer is singular name of WorkloadRebalancer.
+	ResourceSingularWorkloadRebalancer = "workloadrebalancer"
+	// ResourcePluralWorkloadRebalancer is kind plural name of WorkloadRebalancer.
+	ResourcePluralWorkloadRebalancer = "workloadrebalancers"
+	// ResourceNamespaceScopedWorkloadRebalancer indicates if WorkloadRebalancer is NamespaceScoped.
+	ResourceNamespaceScopedWorkloadRebalancer = false
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:resource:path=workloadrebalancers,scope="Cluster",shortName=wr,categories={karmada-io}
+// +kubebuilder:subresource:status
+// +kubebuilder:storageversion
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// WorkloadRebalancer represents the desired behavior and status of a job which can enforces a rescheduling.
+//
+// Notes: make sure the clocks of controller-manager and scheduler are synchronized when using this API.
```

Comment:
```suggestion
```
This note is not clear enough, given it is the basis for a cluster administrator to ensure the clock is consistent between each node. So just remove it to avoid verbosing.